### PR TITLE
Split currentLocation into two elements #1869

### LINKED ISSUE #1869: Split currentLocation into two elements
`currentLocation` is a new element that was introduced with the ALMA FIX Transformation. Since this is a concatinated element one cannot select only a certain element information if the string.
DigiBib asked for it. Also for querying and identifying the sublibrary it can help.
@hagbeck said it is okay for Dortmund if we split the element as long as they receive the info from both elements.

### PR DESCRIPTION
**Pull request by TobiasNx**
Resolves #1869.

This is no structural API-Break but a break concerning the value. The element was introduced when asked by @hagbeck in the beginning of the ALMA-MARC-Transformation. The joined elements showed to be not useful for Digibib. @hagbeck said that as long as they receive both element values they are fine if they are informed when and how this change happens.
 
It was not part of the old ALEPH Morph. I think that this change is okay, when @gregorbg @hagbeck and @blackwinter give their okay.

@acka47 can you have a look if the element name is okay, it is custom due to the naming convention in ALMA:

```
ITM
$w | Current library subfield
$x | Current location subfield
```
https://service-wiki.hbz-nrw.de/pages/viewpage.action?pageId=949911658

`currentLocation` will be kept for `$x` and `currentLibrary` will be added for `$w`.

The naming convention in MARC for 852 which is the corresponding element:
`$b - Sublocation or collection (R) ` -> `currentLibrary`
`$c - Shelving location (R)` -> `currentLocation`
https://www.loc.gov/marc/holdings/hd852.html

`$b - Abteilung der besitzenden Institution (W)`
`$c - Singatur/Standort (W)`
https://www.alma-dach.org/alma-marc/holdings/852/852.html

But only merge when all customers gave okay.

### CODE REVIEW

**Review by gregorbg — APPROVED** [review]
Works for me! Thank you for the consideration, it's an honour to be considered a Lobid "customer" now :D

**Comment by hagbeck** [comment]
This will work for us, too.

**Review by hagbeck — APPROVED** [review]

**Comment by TobiasNx** [comment]
> Isn't `currentLocation` redundant with http://purl.org/lobid/lv#callNumber? I guess, there will be at least an overlap...

If you have a look at the changed data you will see that currentLocation and callNumber have different "values". CallNumber is also based on another subfield(s).

**Review by acka47 — APPROVED** [review]
+1

File: src/main/resources/alma/fix/item.fix [hunk 1]
```diff
@@ -16,7 +16,8 @@ do list(path:"ITM  ", "var": "$i")
     end
     copy_field("$i.H", "@ITM-H.$append")
     copy_field("$i.b", "hasItem[].$last.serialNumber")
-    paste("hasItem[].$last.currentLocation", "$i.w", "~/", "$i.x")
+    copy_field("$i.w","hasItem[].$last.currentLibrary")
+  copy_field("$i.x","hasItem[].$last.currentLocation")
```
Comment (blackwinter) [inline src/main/resources/alma/fix/item.fix:20]: Minor: Indentation.

File: src/main/resources/alma/fix/item.fix [hunk 2]
```diff
@@ -55,7 +56,8 @@ do list(path: "HOL  ", "var": "$i")
           add_field( "hasItem[].$append.test","")
           add_field("hasItem[].$last.label", "lobid Bestandsressource")
           set_array("hasItem[].$last.type[]", "Item","PhysikalischerTitel")
-          paste("hasItem[].$last.currentLocation", "$H52.b", "~/", "$H52.c")
+          copy_field("$H52.b","hasItem[].$last.currentLibrary")
+        copy_field("$H52.c","hasItem[].$last.currentLocation")
```
Comment (blackwinter) [inline src/main/resources/alma/fix/item.fix:60]: Dto.

**Comment by TobiasNx** [comment]
@dr0i can you have a look at the failing test. I am not able to fix it. It should match /src/test/resources/alma-fix/99371314897806441.json

**Comment by TobiasNx** [comment]
also @hagbeck said it would be okay, if the change would be productive on monday.

**Comment by dr0i** [comment]
> @dr0i can you have a look at the failing test. I am not able to fix it. It should match /src/test/resources/alma-fix/99371314897806441.json

You did well by providing this test - it showed, that the queried field wasn't yet configured to be queryable in the elasticsearch index-config.  Fixed with [40f50b0](https://github.com/hbz/lobid-resources/pull/1873/commits/40f50b03c1d5a7355a0b9ba5c1688213977b3b28).

**Review by dr0i — APPROVED** [review]

**Pull request merged**